tests/raftstore/test_transport.rs

```diff
@@ -0,0 +1,106 @@
+use std::sync::Arc;
+use super::cluster::{Cluster, Simulator};
+use super::node::new_node_cluster;
+use super::server::new_server_cluster;
+use super::util::must_get_equal;
+use std::collections::HashSet;
+
+fn split_leader_with_majority(leader: u64, count: u64) -> (HashSet<u64>, HashSet<u64>) {
+    if leader <= (count+1)/2 {
+        ((1..count/2+2).collect(), (count/2+2..count+1).collect())
+    } else {
+        (((count+1)/2..count+1).collect(), (1..(count+1)/2).collect())
+    }
+}
+
+fn split_leader_with_minority(leader: u64, count: u64) -> (HashSet<u64>, HashSet<u64>) {
+    let mut s1 = HashSet::new();
+    let mut s2 = HashSet::new();
+    let mut vec: Vec<u64> = (1..count + 1).collect();
+    vec.swap(0, (leader - 1) as usize);
+    for i in 0..(count - 1) / 2 {
```

Comment:
should be `let s1: HashSet<_> = vec[0..(count-1)/2].into_iter().cloned().collect()`